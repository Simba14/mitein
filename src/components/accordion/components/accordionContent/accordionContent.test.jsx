import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import AccordionContent from './accordionContent';

describe('AccordionContent', () => {
  it('should render correctly', () => {
    const rendered = renderer.create(
      <AccordionContent isOpen ariaId="panel-01" ariaLabelledBy="button-01">
        <div>this is the content</div>
      </AccordionContent>,
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  describe('when is open', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        <AccordionContent isOpen ariaId="panel-01" ariaLabelledBy="button-01">
          <div>this is the content</div>
        </AccordionContent>,
      );
    });

    it('should show the container', () => {
      expect(wrapper.find('.accordionContent')).toHaveLength(1);
    });

    it('should show the content', () => {
      expect(wrapper.hasClass('isOpen')).toEqual(true);
    });
  });

  describe('when is not open', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        <AccordionContent isOpen={false} ariaId="panel-01" ariaLabelledBy="button-01">
          <div>this is the content</div>
        </AccordionContent>,
      );
    });

    it('should not show the content', () => {
      expect(wrapper.hasClass('isOpen')).toEqual(false);
    });
  });
});
