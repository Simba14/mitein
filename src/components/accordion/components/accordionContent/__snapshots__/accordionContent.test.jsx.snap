// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionContent should render correctly 1`] = `
<div
  aria-labelledby="button-01"
  className="accordionContent isOpen"
  id="panel-01"
  role="region"
>
  <div>
    this is the content
  </div>
</div>
`;
